Homework_2/report/main.tex

```diff
@@ -0,0 +1,173 @@
+    0 & \text{drugače}
+\end{cases}
+$$
+in tako zapišemo integral kot
+$$
+\mathbf{F} = \int_{\mathbb{R}^3} \int_{\mathbb{R}^3} K(\mathbf{r_1 - r_2}) \, \chi_{T_1}(\mathbf{r_1}) \, \chi_{T_2}(\mathbf{r_2}) \, d\mathbf{r_1}d\mathbf{r_2}
+$$
+Sedaj uvedemo novo spremenljivko $\mathbf{r} = (x,y,z) = \mathbf{r_1} - \mathbf{r_2}$. Ob upoševanju domen za $\mathbf{r_1}$ in $\mathbf{r_2}$ je $x \in [-3,-1]$, $y \in [-1,1]$ in $z \in [-1,1]$. Jakovijeva matrika je $-I_3$, kjer je $I_3$ enotska matrika v treh dimenzijah, zato je determinanta enaka $-1$. Iz tega sledi $d\mathbf{r_2} = |\text{det} J| \, d\mathbf{r} = d\mathbf{r}$, torej lahko pišemo
+$$
+\mathbf{F} = \int_{\mathbb{R}^3} \chi_{T_1}(\mathbf{r_1}) \int_{\mathbb{R}^3} \chi_{T_2}(\mathbf{r_1} - \mathbf{r}) K(\mathbf{r}) \, d\mathbf{r} \, d\mathbf{r_1}.
+$$
+Vrstni red integracije lahko zamenjamo in upoštevamo
+$$
+\chi_{T_2}(\mathbf{r_1} - \mathbf{r}) = 
+\begin{cases}
+    1 & \text{če } \mathbf{r_1} - \mathbf{r} \in T_2 \\
+    0 & \text{drugače}
+\end{cases}
+=
+\begin{cases}
+    1 & \text{če } \mathbf{r_1} \in T_2 + \mathbf{r} \\
+    0 & \text{drugače}
+\end{cases}
+=
+\chi_{T_2 + \mathbf{r}}(\mathbf{r_1}).
+$$
+Tako dobimo
+$$
+\mathbf{F} = \int_{\mathbb{R}^3} K(\mathbf{r}) \int_{\mathbb{R}^3} \chi_{T_1}(\mathbf{r_1}) \, \chi_{T_2 + \mathbf{r}}(\mathbf{r_1}) \, d\mathbf{r_1} \, d\mathbf{r}.
+$$
+Pomemben razmislek je, da je notranji integral enak volumnu preseka kock $T_1$ in $T_2 + \mathbf{r}$, ki ga označimo z $V(\mathbf{r})$. Končen integral je tako
+$$
+\mathbf{F} = \int_{\mathbb{R}^3} K(\mathbf{r}) \, V(\mathbf{r}) \, d\mathbf{r}.
+$$
+Ker je presek dveh kock kvader, lahko $V(\mathbf{r})$ izračunamo kot produkt presekov po posameznih koordinatah. Ob upoštevanju domen za $\mathbf{r}$ dobimo
+$$
+V(x,y,z) = \left(\text{min}(1,3+x) - \text{max}(0,2+x)\right)(1+|y|)(1+|z|).
+$$
+Upoštevmo še dejstvo, da bo končna sila le v $x$ smeri, saj sta kocki vzporedni. Silo lahko tako zapišemo kot
+$$
+F_x = \int_{-3}^{-1} \int_{-1}^{1} \int_{-1}^{1} \frac{x}{\left(x^2+y^2+z^2\right)^{3/2}} \, V(x,y,z) \, dz \, dy \, dx,
+$$
+kar nam da časovno kompleksnost $O(n^3)$. Kompleksnost lahko še dodatno zmanjšamo na $O(n^2)$, če analitično izračunamo integral po $z$. Potrebno je overdnotiti
+$$
+\int_{-1}^{1} \frac{1-|z|}{\left(x^2+y^2+z^2\right)^{3/2}} \, dz.
+$$
+Ker je integrand soda funkcija, lahko integriramo na intervalu $[0,1]$ in rezultat pomnožimo z 2 (ter uporabimo $|z| = z$). Integral je tako
+$$
+2\int_{0}^{1} \frac{1}{\left(x^2+y^2+z^2\right)^{3/2}} \, dz - 2\int_{0}^{1} \frac{z}{\left(x^2+y^2+z^2\right)^{3/2}} \, dz.
+$$
+Pogledamo v tabele integralov \cite{Bronštejn_Semendjaev_1984}, izračunamo in dobimo
+$$
+\frac{2}{(x^2+y^2)\sqrt{x^2+y^2+1}} - \frac{2}{\sqrt{x^2+y^2}} + \frac{2}{\sqrt{x^2+y^2+1}} = f(x,y).
+$$
+Označimo $l(x) = \left(\text{min}(1,3+x) - \text{max}(0,2+x)\right)$. Končen integral je tako
+$$
+F_x = \int_{-3}^{-1} \int_{-1}^{1} x \, l(x) (1-|y|) f(x,y) \, dy \, dx.
```

Comment:
Tako ali tako uporabite sodost v $y$ zato predlagam, da računate kar

$$
\int_{-3}^1 \int_0^1 x l(x)(1-y)f(x, y)dx dy
$$

in se s tem izognete pralagajanju algoritma.